app/controllers/api_controller.rb

```diff
@@ -40,7 +40,11 @@ def doorkeeper_application_owner
   end
 
   def user_for_paper_trail
-    current_user.owner_id
+    request.headers['X-Current-User']
```

Comment:
Is this the intended? This changes the current behaviour, so if the header is not present (which it won't be), then the user will nil, rather than maintain current logic and set it to the owner_id.

Reply:
This was suggested by @martyn-w, but I'm open to all suggestions.

Maybe I should make the migration go through previous paper trail objects and if the paper trail user is a supplier id, then set the user to nil and backfill the supplier_id field?

Reply:
I am probably out of the loop here, I was more thinking something like `request.headers['X-Current-User'] || current_user.owner_id` so it falls back on current behaviour. But I'm sure @martyn-w will advise :)

Reply:
interesting point, but rather than fall back, I think it will be cleaner to migrate the old values (whodunit) to the new field (supplier_id). So I think this implementation is good 👍